code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -395,3 +394,223 @@
 	component_parts += new /obj/item/stack/sheet/glass(null)
 	component_parts += new cell_type(null)
 	RefreshParts()
+
+// Handheld chem dispenser
+/obj/item/handheld_chem_dispenser
+	name = "handheld chem dispenser"
+	icon = 'icons/obj/chemical.dmi'
+	item_state = "handheld_chem"
+	icon_state = "handheld_chem"
+	flags = NOBLUDGEON
+	var/ui_title = "Handheld Chem Dispenser"
+	var/obj/item/stock_parts/cell/high/cell = null
+	var/amount = 10
+	var/mode = "dispense"
+	var/is_drink = FALSE
+	var/list/dispensable_reagents = list("hydrogen", "lithium", "carbon", "nitrogen", "oxygen", "fluorine",
+	"sodium", "aluminum", "silicon", "phosphorus", "sulfur", "chlorine", "potassium", "iron",
+	"copper", "mercury", "plasma", "radium", "water", "ethanol", "sugar", "iodine", "bromine", "silver", "chromium")
+	var/current_reagent = null
+	var/efficiency = 0.2
+	var/recharge_rate = 1 // Keep this as an integer
+
+/obj/item/handheld_chem_dispenser/New()
+	..()
+	cell = new(src)
+	dispensable_reagents = sortList(dispensable_reagents)
+	current_reagent = pick(dispensable_reagents)
+	update_icon()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/handheld_chem_dispenser/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/handheld_chem_dispenser/get_cell()
+	return cell
+
+/obj/item/handheld_chem_dispenser/afterattack(obj/target, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(!current_reagent)
+		return
+
+	if(!amount)
+		return
+
+	if(!check_allowed_items(target,target_self = TRUE))
+		return
+
+	if(!target.is_refillable())
+		return
+
+	if(mode == "dispense")
+		var/free = target.reagents.maximum_volume - target.reagents.total_volume
+		var/actual = min(amount, cell.charge / efficiency, free)
+		target.reagents.add_reagent(current_reagent, actual)
+		cell.charge -= actual / efficiency
+		if(actual)
+			to_chat(usr, "You dispense [amount] units of [current_reagent] into the [target].")
```

Comment:
Needs spans, and also, since you have the `user` arg passed in, use that over `usr`. Same for the other `to_chat`s below.
```suggestion
			to_chat(user, "<span class='notice'>You dispense [amount] units of [current_reagent] into the [target].</span>")
```